From 0621121b76e2ddc487258f2b38f7cd9d695b7f70 Mon Sep 17 00:00:00 2001
From: Hao Zheng Li <hli@extremenetworks.com>
Date: Wed, 31 Jul 2019 12:53:59 -0400
Subject: [PATCH] proper line handling for complete function

---
 autoload/fzf/vim.vim | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/autoload/fzf/vim.vim b/autoload/fzf/vim.vim
index 35a1143..067c4ce 100644
--- a/autoload/fzf/vim.vim
+++ b/autoload/fzf/vim.vim
@@ -1253,7 +1253,7 @@ function! s:complete_insert(lines)
   if mode() =~ 't'
     call feedkeys('a', 'n')
   else
-    execute "normal! \<esc>la"
+    execute s:eol ? "normal! \<esc>a" : "normal! \<esc>la"
   endif
 endfunction
 
-- 
2.17.1

