From 49c2cf187a5469ed7222ef8e029f98229d666fbe Mon Sep 17 00:00:00 2001
From: Jack Li <jack.lee2980@gmail.com>
Date: Mon, 22 Jul 2019 19:14:12 -0400
Subject: [PATCH] add prefer term option for vim

---
 plugin/fzf.vim | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/plugin/fzf.vim b/plugin/fzf.vim
index 1d1910b..5e235e8 100644
--- a/plugin/fzf.vim
+++ b/plugin/fzf.vim
@@ -387,18 +387,22 @@ try
   endif
 
   let prefer_tmux = get(g:, 'fzf_prefer_tmux', 0)
+  let prefer_term = get(g:, 'fzf_prefer_term', 0)
   let use_height = has_key(dict, 'down') && !has('gui_running') &&
         \ !(has('nvim') || s:is_win || has('win32unix') || s:present(dict, 'up', 'left', 'right', 'window')) &&
         \ executable('tput') && filereadable('/dev/tty')
   let has_vim8_term = has('terminal') && has('patch-8.0.995')
   let has_nvim_term = has('nvim-0.2.1') || has('nvim') && !s:is_win
   let use_term = has_nvim_term ||
-    \ has_vim8_term && !has('win32unix') && (has('gui_running') || s:is_win || !use_height && s:present(dict, 'down', 'up', 'left', 'right', 'window'))
+    \ has_vim8_term && !has('win32unix') && (has('gui_running') || s:is_win || s:present(dict, 'down', 'up', 'left', 'right', 'window'))
   let use_tmux = (!use_height && !use_term || prefer_tmux) && !has('win32unix') && s:tmux_enabled() && s:splittable(dict)
   if prefer_tmux && use_tmux
     let use_height = 0
     let use_term = 0
   endif
+  if prefer_term && use_term
+    let use_height = 0
+  endif
   if use_height
     let height = s:calc_size(&lines, dict.down, dict)
     let optstr .= ' --height='.height
-- 
2.17.1

